csrc/ir/nodes.cpp

```diff
@@ -2111,19 +2111,36 @@ std::vector<PolymorphicValue> MmaOp::evaluate(
 
   // After removing the broadcast dimensions, the format should be
   // [M, K] x [K, N] compatible with aten::matmul format.
-  auto output = in_a.matmul(in_b);
+  at::Tensor output = in_a.matmul(in_b);
 
   // ATen preserves the input dtype whereas MmaOP generates float outputs.
-  // Cast to the dtype of the MmaOp output for consistency.
-  // NOTE: MmaOp returns the float output, whereas in the evaluate method,
-  //      we are casting from float -> input_dtype -> float. This will lead
-  //      to loss of precision.
-  //      MmaOp::evaluate should be modified to effectively handle cast(MmaOp(H,
-  //      H), H) This will avoid the above cast chain and precision issue.
-  if (tv_a->getDataType() != out()->getDataType().value()) {
-    output = output.to(data_type_to_aten(out()->getDataType().value()));
-  }
-  return {output};
+  // If the MmaOp is followed by a cast, skip casting back the input to avoid
+  // roundtrip casts. For eg: MmaOp (H, H) + CastOp (F, H) will result in
+  // F->H->F->H impacting precision and performance. Note: This potentially
+  // binds the MmaOp output in the expression evaluator to a tensor
+  //        of different datatype.
+  const std::vector<Expr*> uses = out()->uses();
```

Comment:
```suggestion
  const std::vector<Expr*>& uses = out()->uses();
```

Otherwise there's an extra copy.